data/ha/hawk_test_driver.py

```diff
@@ -0,0 +1,615 @@
+STONITH_CHKBOX = '//input[contains(@type, "checkbox") and contains(@value, "stonith-sbd")]'
+HREF_CONSTRAINTS = '//a[contains(@href, "#constraints")]'
+HREF_NODES = '//a[contains(@href, "#nodes")]'
+HREF_TAGS = '//a[contains(@href, "#tags")]'
+HREF_ALERTS = '//a[contains(@href, "#alerts")]'
+HREF_FENCING = '//a[contains(@href, "#fencing")]'
+RSC_ROWS = "//*[@id=\"resources\"]/div[1]/div[2]/div[2]/table/tbody/tr"
+DROP_DOWN_FORMAT = "//*[@id=\"resources\"]/div[1]/div[2]/div[2]/table/tbody/tr[%d]/td[6]/div/div"
+STONITH_MAINT_ON = '//a[contains(@href, "stonith-sbd/maintenance_on")]'
+NODE_DETAILS = "//*[@id=\"nodes\"]/div[1]/div[2]/div[2]/table/tbody/tr[1]/td[5]/div/a[2]"
+DISMISS_MODAL = "//*[@id=\"modal\"]/div/div/div[3]/button"
+CLEAR_STATE = "//*[@id=\"nodes\"]/div[1]/div[2]/div[2]/table/tbody/tr[1]/td[5]/div/div/button"
+NODE_MAINT = '//a[contains(@href, "maintenance") and contains(@title, "Switch to maintenance")]'
+NODE_READY = '//a[contains(@href, "ready") and contains(@title, "Switch to ready")]'
+GENERATE_REPORT = "//*[@id=\"generate\"]/form/div/div[2]/button"
+CLONE_CHILD = ('//select[contains(@data-help-filter, ".row.resource") and '
+               'contains(@name, "clone[child]")]')
+TARGET_ROLE_FORMAT = ('//select[contains(@class, "form-control") and '
+                      'contains(@name, "%s[meta][target-role]")]')
+STONITH_MAINT_OFF = ('//a[contains(@href, "stonith-sbd") and '
+                     'contains(@title, "Disable Maintenance Mode")]')
+
+class hawkTestDriverError(Exception):
+    """Base class for exceptions in this module."""
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return repr(self.value)
+
+class hawkTestDriver:
+    def __init__(self, addr='localhost', port=7630, browser='firefox', version='12-SP2'):
+        self.set_addr(addr)
+        self.set_port(port)
+        self.timeout_scale = 1
+        self.set_browser(browser)
+        self.driver = ''
+        self.retval = 0
+        self.test_version = str(version)
+
+    def set_addr(self, addr):
+        if isinstance(addr, str):
+            self.addr = addr
+            return self.addr
+        raise hawkTestDriverError('Unexpected type for host address')
+
+    def set_port(self, port):
+        if isinstance(port, int):
+            self.port = port
+            return self.port
+        raise hawkTestDriverError('Port must be an integer')
+
+    def set_browser(self, browser):
+        if browser in ['chrome', 'chromium', 'firefox']:
+            self.browser = str(browser).lower()
+            if browser == 'firefox':
+                self.timeout_scale = 2.5
+            else:
+                self.timeout_scale = 1
+            return self.browser
+        raise hawkTestDriverError('Browser must be chrome, chromium or firefox')
+
+    def return_value(self, *value):
+        if value:
+            self.retval = int(value[0])
+        return self.retval
```

Comment:
how about set_retval and get_retval methods instead of this mixture method ?

Reply:
Will do.